src/test/java/com/google/javascript/cl2dts/shouldResolveForwardDeclares/forward.js

```diff
@@ -0,0 +1,11 @@
+goog.provide('forward.A');
```

Comment:
Makes sense, but this does sound like a bug in JSCompiler, being unable to use generic types before they are being declared, but also not receiving a warning about it (just JSCompiler things, I guess). Maybe file an issue with them (internal bug tracker)?